### ISSUE TITLE
CI only builds website when pushed to a branch

### ISSUE THREAD

**Issue by cedounet** [hendricius/the-sourdough-framework#349]
Ideally would do the 3 ebook/pdf/website, but at very least the pdf and website as tools are very different.

**Comment by hendricius** [comment]
So not on every push - only when pushed to a branch?

**Comment by cedounet** [comment]
you can do 4 things 
1) push to branch
2) do a PR
3) merge into main
4) push direct to main

1) and 2) will only build website which takes 5 min.  It should also build the pdf, especially as it runs in parallel and will total to about 5 min as well.  ebook is less important as it uses tex4ht as well.

3) Will build a docker image and build pdf(2x)/ebook/b-w/website.  That takes 30 min, cf https://github.com/hendricius/the-sourdough-framework/actions/runs/8775475012

**Comment by hendricius** [comment]
Alrighty - PR opened. Please have a look if I missed something. Hopefully this will bring CI times down.

1. Book + website is tested.
2. Book + website is tested
3. New docker image is built with `latest` tag. Book & website is tested and then released
4. Same as 3

**Comment by hendricius** [comment]
Unfortunately opening up a PR will the workflow, but pushing to a PR will also trigger the workflow. So initially when the PR is opened and not a draft - 2 runs will take place. Subsequent pushes to the PR will only trigger 1 run.

**Comment by cedounet** [comment]
pdf version from CI and website, still is from April 21st despite several push... so there must be some glitch.  But yes it is faster 😸

**Comment by hendricius** [comment]
Now it seems to be releasing properly again 😎
